lib/std/zig/ZonGen.zig

```diff
@@ -411,50 +435,26 @@ fn expr(zg: *ZonGen, node: Ast.Node.Index, dest_node: Zoir.Node.Index) Allocator
                 });
                 try zg.expr(elem_node, @enumFromInt(elem_dest_node));
             }
-        },
-    }
-}
 
-fn parseStrLit(zg: *ZonGen, token: Ast.TokenIndex, offset: u32) !u32 {
-    const raw_string = zg.tree.tokenSlice(token)[offset..];
-    const start = zg.string_bytes.items.len;
-    switch (try std.zig.string_literal.parseWrite(zg.string_bytes.writer(zg.gpa), raw_string)) {
-        .success => return @intCast(start),
-        .failure => |err| {
-            try zg.lowerStrLitError(err, token, raw_string, offset);
-            return error.BadString;
+            const names = zg.extra.items[names_start .. names_start + full.ast.fields.len];
+            outer: for (0..names.len) |i| {
+                for (i + 1..names.len) |j| {
+                    if (names[i] == names[j]) {
+                        const first_field_node = full.ast.fields[i];
+                        const first_token = tree.firstToken(first_field_node) - 2;
+                        const dup_field_node = full.ast.fields[j];
+                        const dup_token = tree.firstToken(dup_field_node) - 2;
+                        try zg.addErrorTokNotes(first_token, "duplicate struct field name", .{}, &.{
+                            try zg.errNoteTok(dup_token, "duplicate name here", .{}),
+                        });
+                        break :outer;
+                    }
+                }
+            }
```

Comment:
O(n^2) logic is not okay -- this should be using a hashmap to do the deduping. I'll do this if you're happy with me pushing to your branch?

Reply:
Sure go for it!